### ISSUE TITLE
Static and SWR rendering not working

### ISSUE THREAD

**Issue by mcrapts** [nuxt/nuxt#18787]
### Environment

Nuxi 3.1.2

RootDir: /home/projects/nuxt-starter-nltug8
Nuxt project info: 

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.1.2`
- Nitro Version:    `2.1.1`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `routeRules`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-nltug8?file=package.json

### Describe the bug

I created 4 pages, each of them showing `new Date()`:

* spa.vue
* ssr.vue
* static.vue
* swr.vue

I created the following routeRules: 
```
export default defineNuxtConfig({
  routeRules: {
    '/spa': { ssr: false },
    '/static': { static: true },
    '/swr': { swr: true },
  },
});
```

After running `npm run build && npm run rpeview`, I get the following results when viewing the source for the following routes:
* spa.vue: not show any date at all. This is correct.
* ssr.vue: show a new date every time on refresh. This is correct.
* static.vue: show the date during build time and not have it change during refresh. This is not working, the date is changed every refresh during runtime, the same as ssr.vue.
* swr.vue: show the same date during the defined TTL, and then have it refresh after that. This is not working, the changed is every refresh during runtime, the same as ssr.vue and static.vue.

So the `static` and `swr` options are not doing anything. 

I realize the functionality is experimental. Is this functionality already supposed to work? I was hoping it would!

**Comment by danielroe** [comment]
This should be resolved by https://github.com/unjs/nitro/pull/906.

**Comment by Bakusedu** [comment]
@mcrapts is this working for you now? I've been having same issue.

**Comment by mcrapts** [comment]
No it doesn't work for me yet with Nuxt 3.2.0 and Nitro 2.2.1.

**Comment by Bakusedu** [comment]
@mcrapts - Apparently, this actually works. We've been using a wrong test case to determine this somehow. Based on the Date() example we've been using, note that for routes that are defined as static, this actually works. Check the static page routes source `View source`, note that the date on the source of the page doesn't match the date on the page / browser:
<img width="1792" alt="Screenshot 2023-02-15 at 15 01 02" src="https://user-images.githubusercontent.com/34550806/219048622-2f3d86d1-2a65-4680-b002-95e89a96ef49.png">
<img width="1792" alt="Screenshot 2023-02-15 at 15 00 20" src="https://user-images.githubusercontent.com/34550806/219048707-9a753491-57ed-4a22-a3ab-acf7c00aad28.png">
Note the difference ? 😃 

And also i had this hydration bug on the console on the static route that's specific to the Date() example, i don't know if you experienced the same:
<img width="1792" alt="Screenshot 2023-02-15 at 15 04 24" src="https://user-images.githubusercontent.com/34550806/219049431-8aef34b5-8774-4c23-925d-218af1c74d5f.png">
This actually means the Date()  on the client causes an override on the initially cached html on the server (which contains the correct date which certifies the static route rule actually works).

Test: https://route-rules-test-nuxt-3.vercel.app/

**Comment by mcrapts** [comment]
Maybe the behavior is different if you deploy to vercel? If I run `npm run build` and start the build locally the source code changes on every refresh for static and SWR.

![image](https://user-images.githubusercontent.com/16231555/219103333-7b4484d5-a40b-40ec-a2e0-a7742e1b6dbb.png)

**Comment by danielroe** [comment]
You need Nitro 2.2+ for this to work. Try updating your project to Nuxt 3.2.

**Comment by mcrapts** [comment]
Indeed `swr: 60` works. But `static: true` still doesn't do anything, so I think this is not completely resolved. Though I agree the performance benefit is already significant with `swr: 1`, so this is a valid workaround for most cases. 

I think at least the documentation could be update to be more specific with working examples and less ambiguous description. Hybrid rendering is mentioned as a [key concept](https://nuxt.com/docs/guide/concepts/rendering) in the documentation, but the documentation is ambiguous with headings such as "Coming in Nuxt 3". What does this mean when Nuxt 3 stable is already released? Is this a sneak preview of an upcoming API in a later version of Nuxt 3? Should we expect this to work or not? Most of the questions here could be addressed by fixing the documentation, which I think is also fine.

**Comment by danielroe** [comment]
great. Let's do that in https://github.com/nuxt/nuxt/pull/18743

**Comment by maximilliangeorge** [comment]
Not sure if this is the right place to ask but should hybrid and/or universal rendering use client side routing? All my Nuxt links are refreshing the page regardless of `routeRules`. I was expecting this on the initial page request, with subsequent requests using Vue Router.

**Comment by danielroe** [comment]
Yes, all navigation should be client-side after the first load. If you are experiencing an issue, please do raise one with a reproduction 🙏

**Comment by maximilliangeorge** [comment]
My bad, I was getting an absolute URL (with https://localhost:3000 prefixed) instead of simply the relative path. It is working now.

**Comment by nieyuyao** [comment]
The `static` is still not work with Nuxt 3.3.3 and Nitro 2.3.2.

**Comment by AdrianFahrbach** [comment]
I got the same problem. Everything seems to be SSR except `prerender: true`. Is this expected and only working with a CDN? The PR #18743 that is linked is merged, yet the documentation doesn't really help me. I'm very confused right now.

@danielroe Is hybrid mode working?

I'm using Nuxt 3.12.3 with Nitro 2.9.7.

**Comment by AdrianFahrbach** [comment]
I added `nitro: { preset: "node-server" }` to the Nuxt config and now `static: true` is working as expected.
SWR is still not working though.

**Comment by AdrianFahrbach** [comment]
I created a separate issue for my problem with a reproduction repo included: #28212
Hope this makes debugging this a bit easier.
